parsec/mca/pins/pins.c

```diff
@@ -31,6 +33,19 @@ void parsec_pins_instrument(struct parsec_execution_stream_s* es,
                             parsec_task_t* task)
 {
     assert( method_flag < PARSEC_PINS_FLAG_COUNT );
+    #if defined(PARSEC_PROF_TRACE)
```

Comment:
It would be cheaper to move this check directly in the PARSEC_PINS macro (in parsec/mca/pins/pins.h) and save one function call for non profile-able events. Here we can keep an assert to make sure we are doing the right thing.